Bug: test_memory_checker failed in Post-test sanity check

### Is it platform specific

generic

### Importance or Severity

Moderate

### Description of the bug

 The `memory_checker` test intermittently fails during post-test sanity checks because monit program services remain in "Initializing" state. The current implementation of `check_monit_running()` only
     verifies that the monit daemon process is running, but does not validate that individual services have completed their first execution cycle.

```
>                   pt_assert(False, "!!!!!!!!!!!!!!!! Post-test sanity check failed: !!!!!!!!!!!!!!!!\n{}"
                              .format(json.dumps(post_failed_results, indent=4, default=fallback_serializer)))
E                   Failed: !!!!!!!!!!!!!!!! Post-test sanity check failed: !!!!!!!!!!!!!!!!
E                   [
E                       {
E                           "failed": true,
E                           "check_item": "monit",
E                           "host": "bjw3-can-7260-14",
E                           "services_status": {
E                               "root-overlay": "OK",
E                               "var-log": "OK",
E                               "bjw3-can-7260-14": "OK",
E                               "rsyslog": "OK",
E                               "routeCheck": "Initializing",
E                               "dualtorNeighborCheck": "Initializing",
E                               "diskCheck": "Initializing",
E                               "container_checker": "Initializing",
E                               "vnetRouteCheck": "Initializing",
E                               "memory_check": "Initializing",
E                               "arp_update_checker": "Initializing",
E                               "controlPlaneDropCheck": "Initializing",
E                               "mgmtOperStatus": "Initializing",
E                               "container_memory_telemetry": "Initializing",
E                               "container_memory_snmp": "Initializing",
E                               "container_memory_otel": "Initializing",
E                               "container_memory_gnmi": "Initializing",
E                               "container_eventd": "Initializing",
E                               "container_memory_bmp": "Initializing"
E                           }
E                       }
E                   ]
```


### Steps to Reproduce

1. Run the test memory_checker/test_memory_checker.py multiple times ( ~10)
2. Multiple times it fails in post sanity with monit monitored service in "Initializing" state which is not a valid state 

### Actual Behavior and Expected Behavior

 ## Actual Behavior

  After monit service is restarted, check_monit_running() returns True  and does NOT validate that services are in ready states. 

  Result:
  - Test proceeds while Program services are still in "Initializing" state
  - Post-test sanity check runs and fails with:
    Service 'routeCheck' status: Initializing (expected: Status ok)
    Service 'diskCheck' status: Initializing (expected: Status ok)
  - Intermittent test failures

## Expected Behavior

  After monit service is restarted, check_monit_running() should wait until all services are in their expected operational states before returning True:
  - Filesystem services: "Accessible"
  - Process services: "Running"
  - Program services: "Status ok"

### Relevant log output

```shell

```

### Output of `show version`

```shell

```

### Attach files (if any)

_No response_

--- a/tests/memory_checker/test_memory_checker.py
+++ b/tests/memory_checker/test_memory_checker.py
@@ -27,7 +27,7 @@ pytestmark = [
 CONTAINER_STOP_THRESHOLD_SECS = 200
 CONTAINER_RESTART_THRESHOLD_SECS = 180
 CONTAINER_CHECK_INTERVAL_SECS = 1
-MONIT_RESTART_THRESHOLD_SECS = 320
+MONIT_RESTART_THRESHOLD_SECS = 400
 MONIT_CHECK_INTERVAL_SECS = 5
 WAITING_SYSLOG_MSG_SECS = 30
 MONIT_MEMORY_CHECK_TIMEOUT = 700
@@ -131,17 +131,29 @@ def restore_monit_config_files(duthost):
 
 
 def check_monit_running(duthost):
-    """Checks whether Monit is running or not.
+    """Checks whether Monit is running with all services in ready state.
 
     Args:
         duthost: The AnsibleHost object of DuT.
 
     Returns:
-        Returns True if Monit is running; Otherwist, returns False.
+        Returns True if all services are ready; Otherwise, returns False.
     """
     monit_services_status = duthost.get_monit_services_status()
     if not monit_services_status:
         return False
+
+    # Validate each service is in expected state
+    for service_name, service_info in monit_services_status.items():
+        service_type = service_info.get("service_type", "")
+        state = service_info.get("service_status", "")
+        if state in ["Not monitored", "OK"]:
+            continue
+        # Check expected states per service type
+        if ((service_type == "Filesystem" and state != "Accessible")
+                or (service_type == "Process" and state != "Running")
+                or (service_type == "Program" and state != "Status ok")):
+            return False
 
     return True
 

